Exercise_3.py

```diff
@@ -17,16 +19,48 @@ def append(self, data):
         Insert a new element at the end of the list.
         Takes O(n) time.
         """
+        newNode = ListNode(data)
+        if self.head == None:
+            self.head = newNode
+            return 
+        endNode = self.head
+        while(endNode):
+            endNode = endNode.next
+        endNode = newNode
         
     def find(self, key):
         """
         Search for the first element with `data` matching
         `key`. Return the element or `None` if not found.
         Takes O(n) time.
         """
+        temp = self.head
+        while (temp != None):
+            if temp.data == key:
+                return True
+        return False
         
     def remove(self, key):
         """
         Remove the first occurrence of `key` in the list.
         Takes O(n) time.
         """
+        temp = self.head
+        if(temp != None):
```

Comment:
You can optimize the code quality here for remove function. Please optimize the code quality.